src/SignalR/clients/ts/signalr/tests/HubConnection.Reconnect.test.ts

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { DefaultReconnectPolicy } from "../src/DefaultReconnectPolicy";
+import { HubConnection, HubConnectionState } from "../src/HubConnection";
+import { JsonHubProtocol } from "../src/JsonHubProtocol";
+
+import { VerifyLogger } from "./Common";
+import { TestConnection } from "./TestConnection";
+import {  PromiseSource } from "./Utils";
+
+describe("auto reconnect", () => {
+    it("is not enabled by default", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const closePromise = new PromiseSource();
+            let onreconnectingCalled = false;
+
+            const connection = new TestConnection();
+            const hubConnection = HubConnection.create(connection, logger, new JsonHubProtocol());
+
+            hubConnection.onclose(() => {
+                closePromise.resolve();
+            });
+
+            hubConnection.onreconnecting(() => {
+                onreconnectingCalled = true;
+            });
+
+            await hubConnection.start();
+
+            // Typically this would be called by the transport
+            connection.onclose!(new Error("Connection lost"));
+
+            await closePromise;
+
+            expect(onreconnectingCalled).toBe(false);
+        },
+        "Connection disconnected with error 'Error: WebSocket closed with status code: 0 ().'.");
+    });
+
+    // it("can be opted into", async () => {
```

Comment:
Some testing coverage areas - Reconnect with negotiate redirects, reconnect with skip negotiate, passing in your own `IReconnectPolicy `instead of just the `DefaultReconnectPolicy`

Reply:
Also being able to successfully reconnect twice.